File: valid-mountain-array/valid-mountain-array.java
class Solution {
    public boolean validMountainArray(int[] arr) {
        if(arr.length <3){
            return false;
        }
        boolean rising=false;
        boolean falling=false;
        for(int i=1; i<arr.length;i++){
            if(arr[i-1]==arr[i]){
                return false;
            }
            if(arr[i-1]<arr[i]){
                if(rising==false && falling==false){
                    rising=true;
                }else if(rising==true && falling==false){
                    continue;
                }else if((rising==true || rising==false ) && falling==true){
                    return false;
                }
            }
            if(arr[i-1]>arr[i]){
                if(rising==true && falling==false){
                    falling=true;
                }else if(falling==false && (rising==true || rising == false)){
                    return false;
                }
            }
        }
        if(rising == true && falling ==false){
            return false;
        }
        return true;
    }
}

File: 978-valid-mountain-array/valid-mountain-array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        if len(arr) < 3:
            return False
        
        s = [False for index in range(len(arr))]
        p = [False for index in range(len(arr))]

        for index in range(len(arr)):
            if index == 0:
                s[index] = True
                continue
            if s[index - 1] == False:
                s[index] = False
                continue
            if arr[index] > arr[index - 1]:
                s[index] = True

        for index in range(len(arr) - 1 , -1, -1):
            if index == len(arr) - 1:
                p[len(arr) - 1] = True
                continue
            if p[index + 1] == False:
                p[index] = False
                continue
            if arr[index] > arr[index + 1]:
                p[index] = True
            
        for index in range(len(arr)):
            if index == 0 or index == len(arr) - 1:
                continue
            if s[index] and p[index]:
                return True
        return False